Bookish/Bookish.DataAccess/BarcodeService.cs

```diff
@@ -22,44 +22,42 @@ public BarcodeService(ILibraryService libraryService)
             this.libraryService = libraryService;
         }
 
-        public IEnumerable<int> GetBarcodeIds(string isbn, int copies, string imageFolder)
+        public IEnumerable<NewCopy> GetNewCopiesWithBarcodes(string isbn, string imageFolder)
         {
-            try
-            {
-                Clean(imageFolder);
-            }
-            catch
-            {
-                // ignored
-            }
-
-            var newCopyIds = libraryService.GetNewCopyIds(isbn, copies).ToList();
-            foreach (var copyId in newCopyIds)
+            var newCopies = libraryService.GetNewCopies(isbn).ToList();
+            foreach (var copy in newCopies)
             {
-                CreateBarcodeImage(copyId, imageFolder);
+                CreateBarcodeImage(copy.bookId, imageFolder);
             }
 
-            return newCopyIds;
+            return newCopies;
         }
 
         private static void CreateBarcodeImage(int copyId, string imageFolder)
         {
+            Clean(imageFolder);
```

Comment:
Now you're doing this for every image which is probably overkill. I'd suggest still calling this from the GetNewCopiesWithBarcodes function 🙂